spec/models/reshare_spec.rb

```diff
@@ -3,15 +3,46 @@
     expect(FactoryGirl.build(:reshare)).to be_valid
   end
 
-  it "requires root" do
-    reshare = FactoryGirl.build(:reshare, root: nil)
-    expect(reshare).not_to be_valid
-  end
+  context "validation" do
+    it "requires root when the author is local" do
```

Comment:
Why can't there be local reshare with no root? What if root post was deleted and reshare author is local?

Reply:
This is only validated on create, so it's fine to have the reshare without root later, but a local user can't create a new reshare without root. We maybe need to change that later for the import and validate that somewhere else, but for now that's the current state.